### ISSUE TITLE
Can an option be added to ignore symbols during word navigation?

### ISSUE THREAD

**Issue by cary-rowen** [mltony/nvda-word-nav#1]
This add-on is really interesting, thank the author for his efforts.
Sometimes we need to read a lot of text content, and we are more concerned about the content itself, the symbols may bring additional interference. Can you add an option in the settings panel so that NVDA no longer reports punctuation and symbols encountered during word navigation?

**Comment by mltony** [comment]
Why you can't turn of punctuation using NVDA+P command?

**Comment by cary-rowen** [comment]
HI,

The symbol level setting of NVDA + P does not apply to Ctrl + Left/Right. Regardless of the symbol level setting, NVDA always reports the symbols in the word navigation process.

[look here](https://github.com/nvaccess/nvda/issues/5133)

When I set the navigation method of the word navigation add-on to "Bulky word navigation", I can ignore the symbols encountered, which is exactly what I want.

Thanks.

**Comment by mltony** [comment]
SO does WordNav work for your test case?

**Comment by cary-rowen** [comment]
Yes, the "Bulky word navigation" in wordnav is what I want, using this option NVDA will not report the symbols in it, but "Notepad++ style" will still report the symbols.

**Comment by mltony** [comment]
That is most likely due to the fact that in NOtepad++ style navigation 
symbols are always spoken as a separate utterance, and I believe that 
this is internal NVDA logic is that when comma or another punctuation 
mark is the only character to be spoken, it is always spoken, no matter 
what's the punctuation level is. I think this is reasonable behavior.

**Comment by cary-rowen** [comment]
In your latest code submission, Set "Left control behavior" to "Default NVDA word navigation" You will find that ctrl + Left/Right becomes unavailable.

**Comment by mltony** [comment]
Ah, thanks for report, just fixed that.

**Comment by cary-rowen** [comment]
Thank you, your add-on is very useful.
For the latest bug, navigation in MS Word with UIA support will become confusing, and NVDA will report selected and unselected states.

**Comment by mltony** [comment]
Thanks for another bug report - I think I fixed the "selected" message 
in MS Word, please try the latest revision. Let me know if you find any 
other bugs!

**Comment by cary-rowen** [comment]
Hi,
Very happy, it's fixed.

**Comment by cary-rowen** [comment]
There is an error in the navigation position in some text input boxes.

For example, the comment input box of github issues.

If it is a multi-line text, we may accidentally return to the first line when navigating the last line.

The same problem exists in the Visual Studio editor.

**Comment by mltony** [comment]
I need more precise step-by-step instructions - I couldn't reproduce it 
in github issue comment field.

I think you meant VSCode - I am aware of this and WordNav is probably 
never going to work correctly in VSCode - due to the way VSCode handles 
edit fields.

**Comment by cary-rowen** [comment]
Enter the following in the comment text box on Github, Please keep a blank line between two lines of text:

---

Apples are delicious.

Bananas are my favorite fruit.

---

step by step:

1. Set WordNav to "Right control behavior": "Notepad++ style navigation";
2. Copy the above two lines of test text to the comment text box of Github;
3. Please press Ctrl + Home to move the input cursor to the beginning of the line;
4. Press "Word Navigation" Right ctrl + Right;
5. Press CTRL + Left when you hear the NVDA report "Bananas";
6. At this point, we cannot hear the blank line between the two lines, nor can we return to the last word in the first line. At this time, the focus seems to be directly back to the beginning of the first line, because we heard the border beep


Of course, not just Notepad++ style has such a problem.
In addition, what I mentioned is not Vs Code, but Visual Studio IDE, The problem is similar to this.

**Comment by mltony** [comment]
Still couldn't reproduce this. I tried in Chrome, Firefox and edge. 
Perhaps try to maximize your window and see if if this is still happening?

**Comment by cary-rowen** [comment]
Hi, 


I tried to maximize the browser window, the problem still exists.
Can you make sure that there are more than two blank lines between the two lines of test text in the comment box?



Can I demonstrate this situation to you by recording?

**Comment by mltony** [comment]
Ok, I managed to reproduce it in Chrome - and I tracked down the problem 
to an underlying issue in NVDA, which I just reported:

https://github.com/nvaccess/nvda/issues/11931

**Comment by cary-rowen** [comment]
Thank you very much for the add-on, it is currently working well. I want to close this issue.

Cheers

**Comment by cary-rowen** [comment]
I think it needs to be brought up again.
NVDA merged [PR#11856](https://github.com/nvaccess/nvda/pull/11856)
Before this, I used "Bulky word navigation" for word navigation to ignore the symbols, but now I force the symbols to be read no matter which algorithm I use, which makes the plug-in less effective.
Do you have any opinions?

**Comment by cary-rowen** [comment]
It is useless to adjust the symbol level of NVDA,
Before this, I used the following configuration, the navigation is very efficient, the former will not read the symbols, the latter will read the symbols, this is my ideal effect, making word-nav a must-have add-on for me,
1. Right control behavior: "Bulky word navigation"
2. Right Control+Windows behavior: "Notepad++ style navigation"
When I need to read the symbols, I only need to press the Windows key on the original basis, which is really great.
Sorry, I don’t know if I made the question clear, thanks again.
